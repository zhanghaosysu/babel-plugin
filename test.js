const path = require('path');
const pluginTester = require('babel-plugin-tester').default;
const fixToFixedPlugin = require('./index');

pluginTester({
  pluginName: 'fixToFixedPlugin',
  plugin: fixToFixedPlugin,
  tests: {
    'nothing changed': {
      code: 'toFixed(2)',
      snapshot: true,
      pluginOptions: { useAccounting: true },
    },
    'NumberLiteral': {
      code: '123.456.toFixed(2)',
      snapshot: true
    }
  },
});
