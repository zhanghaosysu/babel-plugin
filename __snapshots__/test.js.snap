// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixToFixedPlugin NumberLiteral: NumberLiteral 1`] = `

123.456.toFixed(2)

      ↓ ↓ ↓ ↓ ↓ ↓

var fixedToFixed = function fixedToFixed(target, _n) {
  var n = _n || 0;
  return Number(Math.round(target + "e" + n) + "e-" + n).toFixed(n);
};

fixedToFixed(123.456, 2);


`;

exports[`fixToFixedPlugin nothing changed: nothing changed 1`] = `

toFixed(2)

      ↓ ↓ ↓ ↓ ↓ ↓

var fixedToFixed = function fixedToFixed(target, _n) {
  var n = _n || 0;
  return Number(Math.round(target + "e" + n) + "e-" + n).toFixed(n);
};

toFixed(2);


`;
